WooCommerce/src/main/kotlin/com/woocommerce/android/ui/common/wpcomwebview/WPComWebViewFragment.kt

```diff
@@ -0,0 +1,102 @@
+package com.woocommerce.android.ui.common.wpcomwebview
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.View
+import android.webkit.WebChromeClient
+import android.webkit.WebView
+import androidx.core.view.isVisible
+import androidx.navigation.fragment.navArgs
+import com.woocommerce.android.R
+import com.woocommerce.android.databinding.FragmentWpcomWebviewBinding
+import com.woocommerce.android.extensions.navigateBackWithNotice
+import com.woocommerce.android.ui.base.BaseFragment
+import com.woocommerce.android.util.WooLog
+import dagger.hilt.android.AndroidEntryPoint
+import org.wordpress.android.fluxc.network.UserAgent
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.util.StringUtils
+import java.io.UnsupportedEncodingException
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import javax.inject.Inject
+
+private const val WPCOM_LOGIN_URL = "https://wordpress.com/wp-login.php"
+
+/**
+ * This fragments allows loading specific pages from WordPress.com with the current user logged in.
+ * It accepts two parameters:
+ * urlToLoad: the initial URL to load
+ * urlToTriggerExit: optional URL or part of URL to trigger exit with notice when loaded.
+ */
+@AndroidEntryPoint
+class WPComWebViewFragment : BaseFragment(R.layout.fragment_wpcom_webview), UrlInterceptor {
```

Comment:
♻️  Can we make `WPComWebViewFragment` not an `UrlInterceptor` but instead move this to maybe an anonymous class? My concern is that `Fragment` shouldn't be considered as an `UrlInterceptor` as it's not its responsibility, WDYT?
```suggestion
class WPComWebViewFragment : BaseFragment(R.layout.fragment_wpcom_webview) {
```

Reply:
Why do you think so? I really don't see why the fragment shouldn't intercept the URLs, intercepting the URL will trigger a navigation event, so this is one of its responsibilities.

Reply:
I believe `Fragment`s should only be responsible for the presentation layer and intercepting URLs seems not to be coupled with presenting - more with the data layer.

IMO this and `getAuthenticationPostData` should lie in a different class, easy to test without Android framework. I think many things result with navigation event in `Fragment` or its other reaction. If we'll put them directly into fragments, I'm afraid we've got a recipe for "God classes", in precise "God fragments".

I understand this can be debatable and `WPComWebViewFragment` isn't big and/or influential for the whole codebase so I'm okay with eventually leaving this as it is but I have to admit - I'm not a fan 😅 .